Source/Data/Models/Message.swift

```diff
@@ -1,135 +1,170 @@
+        if let timeInt = try values.decodeIfPresent(Int.self, forKey: .time) {
+            time = Date(timeIntSince1970: timeInt)
+        } else {
+            time = nil
         }
-        self.mediaSendRequest <- (map["mediaSendRequest"], CodableTransform<MediaModel>())
-        
-        self.imageURL       <- (map["image"], StringToURLTransform())
-        self.media          <- (map["media"], CodableTransform<MediaObject>())
-        
-        self.buttons        <- map["buttons"]
-        
-        self.carousel       <- map["carousel"]
-        
-        self.quickReplies   <- map["quickReplies"]
-        
-        self.type           <- (map["typeId"], EnumTransform<MessageType>())
-        self.status         <- (map["status"], EnumTransform<MessageStatus>())
-        
-        self.agent          <- map["agent"]
-        
-        self.referrer <- map["referrer"]
+        title = try values.decodeIfPresent(String.self, forKey: .title)
+        message = try values.decodeIfPresent(String.self, forKey: .message)
+        if message?.isEmpty == true {
+            message = nil
+        }
+        imageURL = try values.decodeIfPresent(URL.self, forKey: .image)
+        mediaSendRequest = try values.decodeIfPresent(MediaModel.self, forKey: .mediaSendRequest)
+        media = try values.decodeIfPresent(MediaObject.self, forKey: .media)
+        buttons = try values.decodeIfPresent([MessageButton].self, forKey: .buttons)
+        carousel = try values.decodeIfPresent([Message].self, forKey: .carousel)
+        quickReplies = try values.decodeIfPresent([String].self, forKey: .quickReplies)
+        type = try values.decodeIfPresent(Message.MessageType.self, forKey: .type)
+        status = try values.decodeIfPresent(Message.MessageStatus.self, forKey: .status) ?? .success
+        agent = try values.decodeIfPresent(Agent.self, forKey: .agent)
+        referrer = try values.decodeIfPresent(String.self, forKey: .referrer)
     }
-    
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+
+        try container.encode(id, forKey: .id)
+        try container.encodeIfPresent(uuid, forKey: .uuid)
+        if let time {
+            try container.encode(Int(time.timeIntervalSince1970), forKey: .time)
+        }
+        try container.encodeIfPresent(title, forKey: .title)
+        try container.encodeIfPresent(message, forKey: .message)
+        try container.encodeIfPresent(imageURL, forKey: .imageURL)
+        try container.encodeIfPresent(mediaSendRequest, forKey: .mediaSendRequest)
+        try container.encodeIfPresent(media, forKey: .media)
+        try container.encodeIfPresent(buttons, forKey: .buttons)
+        try container.encodeIfPresent(carousel, forKey: .carousel)
+        try container.encodeIfPresent(quickReplies, forKey: .quickReplies)
+        try container.encodeIfPresent(type, forKey: .type)
+        try container.encodeIfPresent(status, forKey: .status)
```

Comment:
This is causing a regression:

When loading in existing messages, the type results in always being `null`. 
This is caused by the encoding looks like.

Can be solved by:
- Making codingKeys adhere to string
- Add `case type = "typeId"`
- And encode its raw value line: `try container.encodeIfPresent(type?.rawValue, forKey: .type)`
- Decoding already does work with the raw value 👍 
- For consistency and mitigating likewise errors, also do this for the status (but this one is not nullable): `try container.encodeIfPresent(status.rawValue, forKey: .status)`

Reply:
Would be nice to have a test case for this as well

Reply:
Resolved issue and added test in: [abdc62c](https://github.com/parley-messaging/ios-library/pull/53/commits/abdc62c612f16978356da27b6a4bdb894a943535)